### PR TITLE
p4c does not catch clone3 with incorrect clone3 third argument

### PR DESCRIPTION
**Pull request by hesingh**
See https://github.com/p4lang/p4c/issues/1687

The fix adds the check to the bmv2 backend code.

Unit test log is included below.  The testdata file has been moved to a negative test. 

```
hemant@ubuntu:~/p4c/build$ ./p4c-bm2-ss --std p4-16 ../testdata/p4_16_bmv_errors/issue1653-bmv2.p4
../testdata/p4_16_bmv_errors/issue1653-bmv2.p4(30): error: v1 model does not support header bvh0 inside struct for clone3 arg h
    bitvec_hdr bvh0;
               ^^^^
../testdata/p4_16_bmv_errors/issue1653-bmv2.p4(49)
        clone3(CloneType.I2E, 0, h);
                                 ^
hemant@ubuntu:~/p4c/build$ 
```

### CODE REVIEW

File: backends/bmv2/simple_switch/simpleSwitch.cpp [hunk 1]
```diff
@@ -154,6 +154,19 @@ Util::IJson* ExternConverter_clone3::convertExternFunction(
...
+        for (auto f : strct->fields) {
+            auto ftype = f->type;
+            if (ftype->is<IR::Type_Header>()) {
+                ::error("v1 model does not support header %1% inside struct for clone3 arg %2%",
+                        f, exp);
+                return nullptr;
+            }
+        }
+    }
+
```
Comment (jafingerhut) [inline backends/bmv2/simple_switch/simpleSwitch.cpp]: The same restriction applies to resubmit and recirculate field lists, too.  Would it be as straightforward to add such a check for them, too?

Comment (jafingerhut) [reply backends/bmv2/simple_switch/simpleSwitch.cpp]: PSA does clone, recirculate, and resubmit preservation of selected metadata fields in a different way than the current implementation does, and it isn't currently working yet anyway, so nothing working there yet to update.

**Comment by jafingerhut** [comment]
Hemant, I think perhaps this change is a little premature.  Sorry for any part I played in pushing something like this forward, if so.

The changes to P4_16 + v1model architecture for handling the preservation of programmer-selected metadata fields for recirculated/resubmitted/cloned packets sounds like it could be big enough of a change that it is on the schedule for discussion at the next P4 language design workgroup meeting, the first Monday of Feb 2019.

After some more details have been worked out in that forum, and implemented, it seems somewhat likely that all of the P4_16 + v1model arch programs in the testdata/p4_16_samples directory will need to be modified to conform to the new way of doing it.

It also seems likely to me that the restrictions on what fields can be named in such a field_list could become P4_14-specific, and not applicable to P4_16 programs at all.

**Comment by hesingh** [comment]
Andy, no worries at all.  The plan sounds good.

**Pull request closed without merging**